tests/unit_tests/schema/test_column_name_builder.py

```diff
@@ -0,0 +1,68 @@
+import pytest
+
+from exasol_data_science_utils_python.schema.column_name import ColumnName
+from exasol_data_science_utils_python.schema.column_name_builder import ColumnNameBuilder
+from exasol_data_science_utils_python.schema.table_name_impl import TableNameImpl
+
+
+def test_using_empty_constructor():
+    with pytest.raises(TypeError):
+        column_name = ColumnNameBuilder().build()
+
+
+def test_using_constructor_name_only():
+    column_name = ColumnNameBuilder(name="table").build()
+    assert column_name.name == "table" and column_name.table_name is None
+
+
+def test_using_constructor_table():
+    column_name = ColumnNameBuilder(name="table", table_name=TableNameImpl("table")).build()
+    assert column_name.name == "table" and column_name.table_name.name is "table"
+
+
+def test_using_with_name_only():
+    column_name = ColumnNameBuilder().with_name("table").build()
+    assert column_name.name == "table" and column_name.table_name is None
+
+
+def test_using_with_table():
+    column_name = ColumnNameBuilder().with_name("table").with_table_name(TableNameImpl("table")).build()
+    assert column_name.name == "table" and column_name.table_name.name == "table"
+
+
+def test_from_existing_using_with_table():
+    source_column_name = ColumnName("table")
+    column_name = ColumnNameBuilder(column_name=source_column_name).with_table_name(TableNameImpl("table")).build()
+    assert source_column_name.name == "table" \
+           and source_column_name.table_name is None \
+           and column_name.name == "table" \
+           and column_name.table_name.name == "table"
+
+
+def test_from_existing_using_with_name():
+    source_column_name = ColumnName("table", TableNameImpl("table"))
+    column_name = ColumnNameBuilder(column_name=source_column_name).with_name("table1").build()
+    assert source_column_name.name == "table" \
+           and source_column_name.table_name.name == "table" \
+           and column_name.table_name.name == "table" \
+           and column_name.name == "table1"
```

Comment:
if you give both the column and the table  the name "table", you can't be sure if the test is working correctly!
```suggestion
def test_from_existing_using_with_name():
    source_column_name = ColumnName("column", TableNameImpl("table"))
    column_name = ColumnNameBuilder(column_name=source_column_name).with_name("column1").build()
    assert source_column_name.name == "column" \
           and source_column_name.table_name.name == "table" \
           and column_name.table_name.name == "table" \
           and column_name.name == "column1"
```